src/assigner.php

```diff
@@ -14,7 +25,19 @@
     $canUseAssigner = Assigner::canUseAssigner();
     $data["canUseAssigner"] = $canUseAssigner;
 
-    if (isset($_POST["assigner"]) && $canUseAssigner) {
+    // Check if the user is on cooldown
+    $lastUsageTime = $_SESSION['lastAssignerUsage'] ?? 0;
+    $currentTime = time();
+    $timeSinceLastUsage = $currentTime - $lastUsageTime;
+
+    if ($timeSinceLastUsage < $cooldownPeriod) {
+        $data["onCooldown"] = true;
+        $data["cooldownRemaining"] = $cooldownPeriod - $timeSinceLastUsage;
+    } else {
+        $data["onCooldown"] = false;
+    }
```

Comment:
This would be best handled in Assigner class, similarly to `Assigner::canUseAssigner`